analytics/views.py

```diff
@@ -1135,6 +1135,9 @@ def support(request: HttpRequest) -> HttpResponse:
             attach_discount_to_realm(realm, new_discount)
             msg = f"Discount of {realm.string_id} changed to {new_discount} from {current_discount} "
             context["message"] = msg
+            event_time = timezone_now()
+            RealmAuditLog.objects.create(realm=realm, acting_user=request.user, event_type=RealmAuditLog.REALM_DISCOUNT_CHANGED,
```

Comment:
Similiarly make `attach_discount_to_realm` take `acting_user` as an optional argument and create the RealmAuditLog objects there.

Also log the `old_discount` and `new_discount` in `extra_data`. Search for `extra_data` in corporate/lib/stripe.py for examples.